app/gui/src/model/suggestion_database/entry.rs

```diff
@@ -54,6 +54,43 @@ pub struct MissingThisOnMethod(pub String);
 
 
 
+// =====================
+// === QualifiedName ===
+// =====================
+
+im_string_newtype! {
+    /// A single segment of a [`QualifiedName`] of an [`Entry`].
+    QualifiedNameSegment
+}
+
+/// A fully qualified name of an [`Entry`].
+///
+/// A string representation of a fully qualified name can be built by joining the
+/// [`QualifiedName::segments`] using an [`ast::opr::predefined::ACCESS`] character.
+#[derive(Debug, Default, Clone, PartialEq)]
+#[allow(missing_docs)]
+pub struct QualifiedName {
+    pub segments: Vec<QualifiedNameSegment>,
+}
+
+impl<'a> FromIterator<&'a str> for QualifiedName {
+    fn from_iter<T>(iter: T) -> Self
+    where T: IntoIterator<Item = &'a str> {
+        let segments = iter.into_iter().map(|s| s.into()).collect();
+        Self { segments }
+    }
+}
+
+impl<'a> IntoIterator for &'a QualifiedName {
+    type Item = &'a QualifiedNameSegment;
+    type IntoIter = impl Iterator<Item = Self::Item>;
```

Comment:
Use this form only if writing the real type would be really hard. Here you are loosing a lot of info. For example, the Vec iterator is double ended one, you can use `rev` fn on it. Because of your type, this is not possible anymore. Use existential type form only when really needed. Here its not needed.